### PR TITLE
completed week 2 assessment

### PR DESCRIPTION
**Pull request by Lilmcg**
Hi Austin. I had difficulty with this assessment. I flew out of town for my daughter's volleyball tournament and ended up having to do pretty much the entire assessment last night. It didn't give me the time to walk away when I ran into road blocks, brainstorm and then keep trying multiple times (although I still tried my best at that). Also, I couldn't successfully run my tests in the last two coding challenges questions. The message from Jest kept saying "test suite failed to run". It kept finding an error in the LAST LINE of the editor, however that line is EMPTY so it makes no sense! That was super disappointing.

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -16,44 +16,157 @@
...
+const isDivisible = (num) => {
+  if(num % 3 === 0) {
+    return `${num} is divisible by three`
+  } else {
+      return `${num} is not divisible by three`
+  }
+
+// Ran the Test
+// Test failed again
+// Error message: Jest encountered an unexpected token. Arrow pointing out an issue on empty line 121 at bottom of editor where no test or code was written // Makes no sense!
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:64]: Yea I think line 169 is missing a curly bracket and thats throwing it all off.

File: code-challenges.test.js [hunk 2]
```diff
@@ -16,44 +16,157 @@
...
+  // Conditional if the number is evenly divisible by 3
+    // Return a statement that the number is divisible by three
+  // Otherwise return a statement that the number is not divisible by three
+
+const isDivisible = (num) => {
+  if(num % 3 === 0) {
+    return `${num} is divisible by three`
+  } else {
+      return `${num} is not divisible by three`
+  }
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:60]: Looks good!

File: code-challenges.test.js [hunk 3]
```diff
@@ -16,44 +16,157 @@
...
+// Create a local variable eachNoun that maps through the array
+  // Return each value by accessing it's first index and capitalize it using .toUppercase and concatenate the remainder of the word starting at the second index using .substring
+  // Return the local variable eachNoun joined into a string using .join
+
+const capitalArrays = (array) => {
+  let eachNoun = array.map(value => {
+    return value[0].toUppercase() + value.substring(1)
+  })
+  return eachNoun.join(" ")
+}
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:109]: Excellent work!

File: code-challenges.test.js [hunk 4]
```diff
@@ -16,44 +16,157 @@
...
+    for(let i=0; i<firstVowel.length; i++){
+      if(vowelList.indexOf(string[i])) !== -1){
+        return firstVowel.indexOf[i].
+} else {
+        return ("")
+}
+
+// Ran the test
+// Test failed.
+// Error message: Test suite failed to run // Jest encountered an unexpected token. Shows an arrow pointing out an issue on empty line 162 at bottom of editor where no test or code was written // Makes no sense again!
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:172]: yup missing curly bracket

File: code-challenges.test.js [hunk 5]
```diff
@@ -16,44 +16,157 @@
...
+
+const firstVowel = (string) => {
+  let vowelList = 'aeiouAEIOU'
+  let vowelCount = 0
+    for(let i=0; i<firstVowel.length; i++){
+      if(vowelList.indexOf(string[i])) !== -1){
+        return firstVowel.indexOf[i].
+} else {
+        return ("")
+}
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:168]: Your logic is super close! You've got a good idea of what tools to use! getting them to cooperate and knowing how you want the data to come back to you is a difficult job. Good work getting this far!

File: coding-concepts.js [hunk 6]
```diff
@@ -8,40 +8,46 @@
...
 }
-// console.log(greeter("LEARN Student"))
+console.log(greeter("LEARN Student"))
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: Hello LEARN Student
 
+// b) Verify and explain:
+// My answer was wrong. There was no return indicated in the code so the result is undefined
```
Comment (WalkerTexas-coder) [inline coding-concepts.js:29]: Good work recognizing whats going on!

File: interview-questions.md [hunk 7]
```diff
@@ -8,62 +8,64 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 5. What is the difference between a class and an object?
 
   Your answer:
+  Objects are a collection of keys and their associated values. Classes are blueprints of objects.
 
   Researched answer:
-
+  Classes are blueprints of objects that make the objects reusable, and define an executable block of code. An object belongs to a class and are a collection of key  value pairs.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:47]: Good stuff!

**Review by WalkerTexas-coder — APPROVED** [review]
Lillian you did some really good work here! I know your weekend was crazy so I appreciate you bringing in your "A" game. Your code logic and tests look really good and hopefully in the future you can get the to run to see that satisfying green. Well done one concepts and interview Q&A's. This PR is approved.

**Comment by Lilmcg** [comment]
Thank you Austin!

Lil

**Pull request merged**